Source/WebCore/platform/graphics/gstreamer/eme/CDMOpenCDM.cpp

```diff
@@ -400,27 +599,27 @@ void CDMInstanceOpenCDM::removeSessionData(const String& sessionId, LicenseType,
         return;
     }
 
-    KeyStatusVector keys;
-
-    auto result = session->remove();
-    if (result.first) {
-        RefPtr<SharedBuffer> cleanMessage = parseResponseMessage(result.second);
-        if (cleanMessage) {
-            SharedBuffer& initData = session->initData();
-            keys.append(std::pair<Ref<SharedBuffer>, MediaKeyStatus>{initData, MediaKeyStatus::Released});
-            GST_DEBUG("session %s removed, message length %u", sessionId.utf8().data(), cleanMessage->size());
-            callback(WTFMove(keys), cleanMessage.releaseNonNull(), SuccessValue::Succeeded);
+    session->remove([callback = WTFMove(callback)](Session* session, bool success, const String& response) {
+        KeyStatusVector keys;
+        if (success) {
+            RefPtr<SharedBuffer> cleanMessage = parseResponseMessage(response);
+            if (cleanMessage) {
+                SharedBuffer& initData = session->initData();
+                keys.append(std::pair<Ref<SharedBuffer>, MediaKeyStatus>{initData, MediaKeyStatus::Released});
```

Comment:
This is inherited code but I think we should try to copy the init data here as well.